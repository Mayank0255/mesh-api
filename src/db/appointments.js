import { performQuery } from ".";

const getAppointmentQuery = `SELECT
	*
FROM
	appointments
WHERE
	id = $1`;

const createAppointmentQuery = `INSERT INTO appointments (type, date, notes, member_id, building_id, request_id, acuity_id)
	VALUES($1, $2, $3, $4, $5, $6, $7)
RETURNING
	*`;

export async function getAppointment(id) {
	return performQuery(getAppointmentQuery, [id]);
}

export async function createAppointment(appointment) {
	return performQuery(createAppointmentQuery, [
		appointment.type,
		appointment.date,
		appointment.notes,
		appointment.member_id,
		appointment.building_id,
		appointment.request_id,
		appointment.acuity_id
	]);
}
