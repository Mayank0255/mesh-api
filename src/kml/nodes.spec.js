import { performQuery } from '../db';
jest.mock('../db');
import { getNodesKML } from './nodes';

describe('getNodesKML', () => {
  it('generates KML for all the nodes', async () => {
    const nodes = [
      {
        id: 9999,
        status: "active",
        create_date: new Date("2019-12-21"),
        device_types: [
          { name: "Omni" },
          { name: "LBE" },
        ],
        panoramas: [
          { url: "https://imgur.com/9999" },
        ],
        lat: '33.123414',
        lng: '-71.123413',
        alt: '230m',
      },
      {
        id: 9123,
        name: "Saratoga",
        status: "active",
        create_date: new Date("2016-02-01"),
        device_types: [
          { name: "LiteAP" },
          { name: "AF24" },
        ],
        panoramas: [],
        lat: '33.953245',
        lng: '-74.123410',
        alt: '80m',
      },
      {
        id: 1013,
        status: "inactive",
        create_date: new Date("2014-08-15"),
        device_types: [],
        notes: "hub node",
        panoramas: [
          { url: "https://imgur.com/1013/1" },
          { url: "https://imgur.com/1013/2" },
        ],
        lat: '31.234524',
        lng: '-75.934582',
        alt: '122m',
      },
    ];

    // getNodes
    performQuery.mockResolvedValueOnce(nodes);

    // getLinks
    performQuery.mockResolvedValueOnce([
      {
        id: 91,
        node_a: nodes[0],
        node_b: nodes[1],
        device_type_a: { name: "LBE" },
        device_type_b: { name: "LiteAP" },
        status: "active",
      },
      {
        id: 92,
        node_a: nodes[2],
        node_b: nodes[1],
        device_type_a: { name: "Unknown" },
        device_type_b: { name: "LiteAP" },
        status: "inactive",
      },
    ]);

    expect(await getNodesKML()).toMatchSnapshot();
  });
});
