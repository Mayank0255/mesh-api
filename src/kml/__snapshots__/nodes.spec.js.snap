// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNodesKML generates KML for all the nodes 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<kml xmlns=\\"http://www.opengis.net/kml/2.2\\">
    <Document>
        <Style id=\\"supernode\\">
    <IconStyle>
        <scale>0.6</scale> 
        <Icon>
            <href>https://i.imgur.com/GFd364p.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,<Style id=\\"hub\\">
    <IconStyle>
        <scale>0.6</scale> 
        <Icon>
            <href>https://i.imgur.com/dsizT9e.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,<Style id=\\"omni\\">
    <IconStyle>
        <scale>0.6</scale> 
        <Icon>
            <href>https://i.imgur.com/dsizT9e.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,<Style id=\\"node\\">
    <IconStyle>
        <scale>0.5</scale> 
        <Icon>
            <href>https://i.imgur.com/OBBZi9E.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,<Style id=\\"hubLink\\">
    <LineStyle>
        <color>ff00ffff</color>
        <width>3</width>
    </LineStyle>
    <PolyStyle>
        <color>00000000</color>
    </PolyStyle>
</Style>,<Style id=\\"backboneLink\\">
    <LineStyle>
        <color>ff00ffff</color>
        <width>3</width>
    </LineStyle>
    <PolyStyle>
        <color>00000000</color>
    </PolyStyle>
</Style>,<Style id=\\"activeLink\\">
    <LineStyle>
        <color>ff0000ff</color>
        <width>3</width>
    </LineStyle>
    <PolyStyle>
        <color>00000000</color>
    </PolyStyle>
</Style>,<Folder>
					<name>1013</name>
					<Placemark>
	<name>Node 1013</name>
		<ExtendedData>
		
		<Data name=\\"Status\\">
    <value>inactive</value>
</Data>
		<Data name=\\"Installed\\">
    <value>Fri Aug 15 2014</value>
</Data>
		<Data name=\\"Devices\\">
    <value></value>
</Data>
		<Data name=\\"Notes\\">
    <value>hub node</value>
</Data>
		<Data name=\\"Links\\">
    <value><a href=\\"https://dashboard.nycmesh.net/requests/1013\\" style=\\"margin-right: 1rem;\\">Dashboard →</a> <a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=1013\\" style=\\"margin-right: 1rem;\\">Tickets →</a></value>
</Data>
		<Data name=\\"Panos\\">
    <value><img src=\\"https://imgur.com/1013/1\\" style=\\"width: 32rem; display: block;\\" /><img src=\\"https://imgur.com/1013/2\\" style=\\"width: 32rem; display: block;\\" /></value>
</Data>
	</ExtendedData>
	<Point>
		<altitudeMode>absolute</altitudeMode>
		<coordinates>-75.934582,31.234524,122m</coordinates>
	</Point>
	<styleUrl>#hub</styleUrl>
</Placemark>
					<Placemark>
	<name>Link</name>
	<ExtendedData>
		<Data name=\\"ID\\">
    <value>92</value>
</Data>
		<Data name=\\"Status\\">
    <value>inactive</value>
</Data>
		<Data name=\\"From\\">
    <value>1013 Unknown Device</value>
</Data>
		<Data name=\\"To\\">
    <value>Saratoga LiteAP</value>
</Data>
	</ExtendedData>
	<LineString>
		<altitudeMode>absolute</altitudeMode>
		<coordinates>-75.934582,31.234524,122m -74.123410,33.953245,80m</coordinates>
	</LineString>
	<styleUrl>#activeLink</styleUrl>
</Placemark>

				</Folder>,<Folder>
					<name>9123</name>
					<Placemark>
	<name>Saratoga</name>
		<ExtendedData>
		<Data name=\\"Name\\">
    <value>Saratoga</value>
</Data>
		<Data name=\\"Status\\">
    <value>active</value>
</Data>
		<Data name=\\"Installed\\">
    <value>Mon Feb 01 2016</value>
</Data>
		<Data name=\\"Devices\\">
    <value>LiteAP, AF24</value>
</Data>
		
		<Data name=\\"Links\\">
    <value><a href=\\"https://dashboard.nycmesh.net/requests/9123\\" style=\\"margin-right: 1rem;\\">Dashboard →</a> <a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=9123\\" style=\\"margin-right: 1rem;\\">Tickets →</a></value>
</Data>
		<Data name=\\"Panos\\">
    <value></value>
</Data>
	</ExtendedData>
	<Point>
		<altitudeMode>absolute</altitudeMode>
		<coordinates>-74.123410,33.953245,80m</coordinates>
	</Point>
	<styleUrl>#node</styleUrl>
</Placemark>
					
				</Folder>,<Folder>
					<name>9999</name>
					<Placemark>
	<name>Node 9999</name>
		<ExtendedData>
		
		<Data name=\\"Status\\">
    <value>active</value>
</Data>
		<Data name=\\"Installed\\">
    <value>Sat Dec 21 2019</value>
</Data>
		<Data name=\\"Devices\\">
    <value>Omni, LBE</value>
</Data>
		
		<Data name=\\"Links\\">
    <value><a href=\\"https://dashboard.nycmesh.net/requests/9999\\" style=\\"margin-right: 1rem;\\">Dashboard →</a> <a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=9999\\" style=\\"margin-right: 1rem;\\">Tickets →</a></value>
</Data>
		<Data name=\\"Panos\\">
    <value><img src=\\"https://imgur.com/9999\\" style=\\"width: 32rem; display: block;\\" /></value>
</Data>
	</ExtendedData>
	<Point>
		<altitudeMode>absolute</altitudeMode>
		<coordinates>-71.123413,33.123414,230m</coordinates>
	</Point>
	<styleUrl>#omni</styleUrl>
</Placemark>
					<Placemark>
	<name>Link</name>
	<ExtendedData>
		<Data name=\\"ID\\">
    <value>91</value>
</Data>
		<Data name=\\"Status\\">
    <value>active</value>
</Data>
		<Data name=\\"From\\">
    <value>9999 LBE</value>
</Data>
		<Data name=\\"To\\">
    <value>Saratoga LiteAP</value>
</Data>
	</ExtendedData>
	<LineString>
		<altitudeMode>absolute</altitudeMode>
		<coordinates>-71.123413,33.123414,230m -74.123410,33.953245,80m</coordinates>
	</LineString>
	<styleUrl>#activeLink</styleUrl>
</Placemark>

				</Folder>
    </Document>
</kml>"
`;
